### PR TITLE
Whitespace homogenizer

### PR DESCRIPTION
**Pull request by zedlopez**
re-opening, using binarysearch

### CODE REVIEW

File: inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t [hunk 1]
```diff
@@ -319,6 +322,17 @@ to document all of that.
         glk_put_char_stream(gg_commandstr, 10); ! newline
     }
   .KPContinue;
+
+    uwl = UnicodeWhitespace + WORDSIZE;
+    for ( ix = 1 : ix <= (a_buffer-->0) : ix++ )
+      if (((a_buffer-->ix) <= 32) || ((((a_buffer-->ix) >= 8192) && ((a_buffer-->ix) <= 8202))))
+        a_buffer-->ix = 32;
+      else {
+        found = a_buffer-->ix;
```
Comment (curiousdannii) [inline inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t:331]: Put this before any of the comparisons. The if above would be retrieving it three times cause the compiler's not smart enough to extract that code.

File: inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t [hunk 2]
```diff
@@ -319,6 +322,17 @@ to document all of that.
         glk_put_char_stream(gg_commandstr, 10); ! newline
     }
   .KPContinue;
+
+    uwl = UnicodeWhitespace + WORDSIZE;
```
Comment (curiousdannii) [inline inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t:326]: Doesn't this mean it skips 133?

Comment (zedlopez) [reply inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t:326]: yeah, remnant of when the length was the first element.

File: inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t [hunk 3]
```diff
@@ -319,6 +322,17 @@ to document all of that.
...
+    for ( ix = 1 : ix <= (a_buffer-->0) : ix++ )
+      if (((a_buffer-->ix) <= 32) || ((((a_buffer-->ix) >= 8192) && ((a_buffer-->ix) <= 8202))))
```
Comment (curiousdannii) [inline inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t:328]: The binary search would almost certainly be faster than the checks for 8192-8202, but then you have to list them all in the array.

**Comment by ganelson** [comment]
My only concern about the use of @binarysearch was whether Inter-to-C supported it, which I'd forgotten - Inter-to-C does not support every Glulx opcode. But in fact it does, and all C test cases pass with the above PR in place.

**Comment by zedlopez** [comment]
(I do run the test suite before every PR.)

**Comment by zedlopez** [comment]
So I was looking at VM_Tokenise again and, um, it seems like it makes more sense for this to happen at the top of the routine before the attempt to look for ' ' characters on which to split words. What do you think, @curiousdannii ?

**Pull request merged**